### PR TITLE
IWCE-2 Created base component structure

### PR DESCRIPTION
**Pull request by veronikaTailored**

### CODE REVIEW

File: src/admin/config.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    <fieldset label="COM_ADMINTASK" name="admintask">
+
+    </fieldset>
+
+    <fieldset name="permissions" description="JCONFIG_PERMISSIONS_DESC" label="JCONFIG_PERMISSIONS_LABEL">
+        <field name="rules" type="rules" component="com_admintask" class="inputbox" filter="rules" validate="rules" label="JCONFIG_PERMISSIONS_LABEL" section="component" />
+	</fieldset>
+
+    <fieldset name="component">
+        <field name="save_history" type="radio" class="btn-group btn-group-yesno" default="0" label="JGLOBAL_SAVE_HISTORY_OPTIONS_LABEL" description="JGLOBAL_SAVE_HISTORY_OPTIONS_DESC">
```
Comment (gunjanpatel) [inline src/admin/config.xml]: What is the use of this config param?

File: src/admin/config.xml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            <option value="0">JNO</option>
+            <option value="1">JYES</option>
+        </field>
+
+        <field name="history_limit" type="text" filter="integer" label="JGLOBAL_HISTORY_LIMIT_OPTIONS_LABEL" description="JGLOBAL_HISTORY_LIMIT_OPTIONS_DESC" default="5"/>
```
Comment (gunjanpatel) [inline src/admin/config.xml]: Also please explain the use of this config option.

File: src/admin/controller.php [hunk 3]
```diff
@@ -0,0 +1,39 @@
+<?php
+
```
Comment (gunjanpatel) [inline src/admin/controller.php]: Remove this extra empty line and also from all the files.

File: src/admin/controller.php [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+/**
+ * @version    CVS: 1.0.0
```
Comment (gunjanpatel) [inline src/admin/controller.php]: What is `CVS` here?

File: src/admin/controller.php [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+ * @package    Com_Admintask
+ * @author     Tasolglobal <testing@tasolglobal.com>
```
Comment (gunjanpatel) [inline src/admin/controller.php]: Please replace the `Tasolglobal <testing@tasolglobal.com>` with `__COMPONENT_CONTACT__` in all files.

File: src/admin/controller.php [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+ * @copyright  2016 Tasolglobal
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+// No direct access
+defined('_JEXEC') or die;
+
+/**
+ * Class AdmintaskController
+ *
+ * @since  1.6
```
Comment (gunjanpatel) [inline src/admin/controller.php]: use `1.0` as a version number as we haven't did any released.

Comment (gunjanpatel) [inline src/admin/controller.php]: Do it on all the files, `@since    1.0`

File: src/admin/controllers/task.php [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+
+jimport('joomla.application.component.controllerform');
+
+/**
+ * Task controller class.
+ *
+ * @since  1.6
+ */
+
+
```
Comment (gunjanpatel) [inline src/admin/controllers/task.php]: Remove this extra lines

File: src/admin/controllers/task.php [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+ *  Constructor
+ *
+ * @throws Exception
+*/
+
+public function __construct()
+{
+		$this->view_list = 'tasks';
+		parent::__construct();
+}
```
Comment (gunjanpatel) [inline src/admin/controllers/task.php]: Code indention is required

File: src/admin/controller.php [hunk 9]
```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * @version    CVS: 1.0.0
+ * @package    Com_Admintask
+ * @author     Tasolglobal <testing@tasolglobal.com>
+ * @copyright  2016 Tasolglobal
```
Comment (gunjanpatel) [inline src/admin/controller.php]: Change from `@copyright  2016 Tasolglobal` to `@copyright  __COMPONENT_COPYRIGHT__` in all files.

File: src/admin/helpers/admintask.php [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+	 */
+
+public static function addSubmenu($vName = '')
+{
+	JHtmlSidebar::addEntry(
+	JText::_('COM_ADMINTASK_TITLE_TASKS'),
+		'index.php?option=com_admintask&view=tasks',
+		$vName == 'tasks'
+		);
+}
```
Comment (gunjanpatel) [inline src/admin/helpers/admintask.php]: Check code indention in all the files.

File: src/admin/config.xml (hunk 1 above)
Comment (veronikaTailored) [reply src/admin/config.xml]: Config param is use for the store the global parameters value in database for that component .

File: src/admin/config.xml (hunk 2 above)
Comment (veronikaTailored) [reply src/admin/config.xml]: -->Config option is use for the global configuration setting .

Comment (gunjanpatel) [reply src/admin/config.xml]: But where? and why?

File: src/admin/config.xml (hunk 1 above)
Comment (gunjanpatel) [reply src/admin/config.xml]: What history you are storing? what is the use of this param?

Comment (veronikaTailored) [reply src/admin/config.xml]: -->history store 1 or 0  and if we set yes that save automatically old versions of item .

File: src/admin/config.xml (hunk 2 above)
Comment (veronikaTailored) [reply src/admin/config.xml]: -->Currently i have not used config in this component .

File: src/admin/helpers/admintask.php [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+	/**
+		* Gets the files attached to an item
+		*
+		* @param   int     $pk     The item's id
+		*
+		* @param   string  $table  The table's name
+		*
+		* @param   string  $field  The field's name
+		*
+		* @return  array  The files
```
Comment (gunjanpatel) [inline src/admin/helpers/admintask.php]: Code indention looks wrong.

File: src/admin/helpers/admintask.php [hunk 12]
```diff
@@ -0,0 +1,81 @@
...
+	{
+			$db = JFactory::getDbo();
+			$query = $db->getQuery(true);
+			$query
+			->select($field)
+			->from($table)
+			->where('id = ' . (int) $pk);
+			$db->setQuery($query);
+
+			return explode(', ', $db->loadResult());
```
Comment (gunjanpatel) [inline src/admin/helpers/admintask.php]: same here. Code indention doesn't looks correct.

File: src/admin/sql/uninstall.mysql.utf8.sql [hunk 13]
```diff
@@ -0,0 +1,3 @@
+DROP TABLE IF EXISTS `#__admintask_tasks`;
+
+DELETE FROM `#__content_types` WHERE (type_alias LIKE 'com_admintask.%');
```
Comment (gunjanpatel) [inline src/admin/sql/uninstall.mysql.utf8.sql]: Why you need this DROP query?

File: src/admin/tables/task.php [hunk 14]
```diff
@@ -0,0 +1,283 @@
...
+class AdmintaskTabletask extends JTable
+{
+	/**
+	 * Constructor
+	 *
+	 * @param   JDatabase  &$db  A database connector object
+	 */
+	public function __construct(&$db)
+	{
+		JObserverMapper::addObserverClassToClass('JTableObserverContenthistory', 'AdmintaskTabletask', array('typeAlias' => 'com_admintask.task'));
```
Comment (gunjanpatel) [inline src/admin/tables/task.php]: Why do you need this?

File: src/admin/tables/task.php [hunk 15]
```diff
@@ -0,0 +1,283 @@
...
+	 *
+	 * @param   array  $array   Named array
+	 * @param   mixed  $ignore  Optional array or list of parameters to ignore
+	 *
+	 * @return  null|string  null is operation was satisfactory, otherwise returns an error
+	 *
+	 * @see     JTable:bind
+	 * @since   1.5
+	 */
+	public function bind($array, $ignore = '')
```
Comment (gunjanpatel) [inline src/admin/tables/task.php]: Do you really think that you need bind method?

File: src/admin/tables/task.php [hunk 16]
```diff
@@ -0,0 +1,283 @@
...
+	 * Returns the parent asset's id. If you have a tree structure, retrieve the parent's id using the external key field
+	 *
+	 * @param   JTable   $table  Table name
+	 * @param   integer  $id     Id
+	 *
+	 * @see JTable::_getAssetParentId
+	 *
+	 * @return mixed The id on success, false on failure.
+	 */
+	protected function _getAssetParentId(JTable $table = null, $id = null)
```
Comment (gunjanpatel) [inline src/admin/tables/task.php]: Why do you need this method?

File: src/media/css/form.css [hunk 17]
```diff
@@ -0,0 +1,44 @@
+.front-end-edit ul {
```
Comment (gunjanpatel) [inline src/media/css/form.css]: Where did you use this css?

File: src/admin/sql/uninstall.mysql.utf8.sql (hunk 13 above)
Comment (veronikaTailored) [reply src/admin/sql/uninstall.mysql.utf8.sql]: Not need now ..i will remove it

File: src/admin/tables/task.php (hunk 14 above)
Comment (veronikaTailored) [reply src/admin/tables/task.php]: No currently not need now .

File: src/admin/tables/task.php (hunk 15 above)
Comment (veronikaTailored) [reply src/admin/tables/task.php]: Bind method not need not..But its need in ACL so i have not remove that .

File: src/media/css/form.css (hunk 17 above)
Comment (veronikaTailored) [reply src/media/css/form.css]: Not used ..

File: src/admin/tables/task.php (hunk 15 above)
Comment (gunjanpatel) [reply src/admin/tables/task.php]: This doesn't need at all. Can you explain what you have changed in this method and what is new in this method which is not in it's parent method?

File: src/admin/helpers/admintask.php [hunk 18]
```diff
@@ -33,16 +33,14 @@ public static function addSubmenu($vName = '')
...
-	*/
+	 * Gets the files attached to an item
+	 *
+	 * @param   int     $pk     The item's id
+	 * @param   string  $table  The table's name
+	 * @param   string  $field  The field's name
+	 *
+	 * @return  array  The files
+	 */
 	public static function getFiles($pk, $table, $field)
```
Comment (gunjanpatel) [inline src/admin/helpers/admintask.php]: Get files? For what? Which file you are getting using this method?

File: src/admin/tables/task.php [hunk 19]
```diff
@@ -27,257 +27,4 @@ public function __construct(&$db)
 		parent::__construct('#__admintask_tasks', 'id', $db);
```
Comment (gunjanpatel) [inline src/admin/tables/task.php:25]: I have asked you why you need `JObserverMapper::addObserverClassToClass` this line of code? Do you explain me why you have used this?

**Comment by gunjanpatel** [comment]
@veronikaTailored you must know what code you are writing. Most of the developers including me copy paste the code but we must know what we have added and why!

**Pull request merged**